pkg/descheduler/controllers/migration/arbitrator/sort.go

```diff
@@ -0,0 +1,190 @@
+func SortJobsByPod(sorter func(pods []*corev1.Pod)) SortFn {
+	return func(jobs []*v1alpha1.PodMigrationJob, podOfJob map[*v1alpha1.PodMigrationJob]*corev1.Pod) []*v1alpha1.PodMigrationJob {
+		var pods []*corev1.Pod
+		jobOfPod := map[*corev1.Pod]*v1alpha1.PodMigrationJob{}
+		rankOfJob := map[*v1alpha1.PodMigrationJob]int{}
+		for _, job := range jobs {
+			pod := podOfJob[job]
+			if pod == nil {
+				rankOfJob[job] = math.MinInt
+				continue
+			}
+			pods = append(pods, pod)
+			jobOfPod[pod] = job
+		}
+
+		// using PodSorter to sort
+		sorter(pods)
+
+		for i, pod := range pods {
+			rankOfJob[jobOfPod[pod]] = i
+		}
+
+		sort.SliceStable(jobs, func(i, j int) bool {
+			return rankOfJob[jobs[i]] < rankOfJob[jobs[j]]
+		})
+		return jobs
+	}
+}
+
+// SortJobsByCreationTime returns a SortFn that stably sorts PodMigrationJobs by create time.
+func SortJobsByCreationTime() SortFn {
+	return func(jobs []*v1alpha1.PodMigrationJob, podOfJob map[*v1alpha1.PodMigrationJob]*corev1.Pod) []*v1alpha1.PodMigrationJob {
+		// calculate the time interval between the creation of migration job and the current for each job.
+		timeMap := map[*v1alpha1.PodMigrationJob]int64{}
+		for _, job := range jobs {
+			timeMap[job] = job.GetCreationTimestamp().Unix()
+		}
+		// perform stable sorting.
+		sort.SliceStable(jobs, func(i, j int) bool {
+			return timeMap[jobs[i]] > timeMap[jobs[j]]
+		})
+		return jobs
+	}
+}
+
+// SortJobsByMigrationStatus returns a SortFn that.
+func SortJobsByMigrationStatus(c client.Client) SortFn {
+	return func(podMigrationJobs []*v1alpha1.PodMigrationJob, podOfJob map[*v1alpha1.PodMigrationJob]*corev1.Pod) []*v1alpha1.PodMigrationJob {
+		// get owner of jobs
+		owners := map[types.UID]*metav1.OwnerReference{}
+		for _, job := range podMigrationJobs {
+			owner, ok := getJobControllerOfPod(podOfJob[job])
+			if !ok {
+				continue
+			}
+			owners[owner.UID] = owner
+		}
+
+		// get all migrating owners
+		migratingOwners := map[types.UID]bool{}
+		for uid := range owners {
+			opts := &client.ListOptions{FieldSelector: fields.OneTermEqualSelector(fieldindex.IndexPodByOwnerRefUID, string(uid))}
+			podList := &corev1.PodList{}
+			err := c.List(context.TODO(), podList, opts, utilclient.DisableDeepCopy)
+			if err != nil {
+				klog.ErrorS(err, "failed to list pods by IndexPodByOwnerRefUID", "OwnerRefUID", uid)
+				continue
+			}
+			for _, pod := range podList.Items {
+				opts = &client.ListOptions{FieldSelector: fields.OneTermEqualSelector(fieldindex.IndexJobByPodUID, string(pod.UID))}
+				jobList := &v1alpha1.PodMigrationJobList{}
+				err = c.List(context.TODO(), jobList, opts, utilclient.DisableDeepCopy)
+				if err != nil {
+					klog.ErrorS(err, "failed to list jobs by IndexJobByPodUID", "pod", klog.KObj(&pod))
+					continue
+				}
+				for _, job := range jobList.Items {
+					if isJobMigrating(&job) {
+						migratingOwners[uid] = true
+					}
+				}
+			}
```

Comment:
Counting migration jobs by pod owner and ordering by counter rather than phase should work better.
And This loop code should turn into a function

Reply:
This func, I want to gather PMJs that belong to same job.

Reply:
I want to give the PMJs more priority if the job that they belong to has some migrating PMJs.